File: app/src/main/java/com/example/superclass/MainActivity.java
 package com.example.superclass;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.example.superclass.gif.GifActivity;
import com.example.superclass.image.ImageOptActivity;

 public class MainActivity extends AppCompatActivity {

    /**
     * 这个程序负责记录
     */


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startActivity(new Intent(this,GifActivity.class));


    }

}

File: pluginapp/src/main/java/com/example/pluginapp/MainActivity.kt
package com.example.pluginapp

import android.content.Intent
import android.os.Bundle

class MainActivity : BaseActivity() {

    override fun onCreate(saveInstanceState: Bundle?) {
        super.onCreate(saveInstanceState)
        setContentView(R.layout.activity_main)


        findViewById(R.id.btn).setOnClickListener {
            //由于是加载类 所有没有注册上下文 只能使用传过来的that
             var intent= Intent(that,SecondActivity::class.java)

            startActivity(intent)
        }
    }
}
